walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/EventTest.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.PriorityQueue;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
+@RunWith(JUnit4.class)
+public final class EventTest {
+  private static final int DURATION_30_MINUTES = 30;
+  private static final int DURATION_60_MINUTES = 60;
+
+  private static final int TIME_0830 = TimeRange.getTimeInMinutes(8, 30);
+  private static final int TIME_1330 = TimeRange.getTimeInMinutes(13, 30);
+
+  private static final Event EVENT_A = new Event(
+      "Event A", TimeRange.fromStartDuration(TIME_0830, DURATION_30_MINUTES), Arrays.asList());
+  private static final Event EVENT_B = new Event(
+      "Event B", TimeRange.fromStartDuration(TIME_1330, DURATION_60_MINUTES), Arrays.asList());
+
+  @Test
+  public void orderEventByStartTime() {
```

Comment:
nit: this is a somewhat advanced critique of your work, which is good. But I think it can be better!

First, I think it would be better to create the two events inside the test (using the other constants is fine). It's not obvious now, but if you build up a lot of tests (which can/should happen in a real project), there will be a large separation between where you define e.g. `EVENT_A` and where you use it. 

Second, the names `EVENT_A` and `EVENT_B` don't make completely obvious that they should be ordered that way. You could name them `earlyEvent` and `lateEvent`, or `morningEvent` and `afternoonEvent`, or even `eventAt0830` and `eventAt1330`, though it might be better to avoid using numbers in variable names.

Finally, it's best to follow the order of "arrange, act, assert": you should first arrange the ingredients for the test, then execute the code you want to test, then check that the outcome is as expected. So I would move the construction of the expected output to the end, just after you call `toArray(...)`.

Reply:
This TotT episode is somewhat relevant: https://drive.google.com/file/d/15GGGANZoaG9JIyGyrTkZWkmYlaih01MJ/view

Reply:
Ah, I didn't see Ali's response before I sent this. No worries!